pkg/operator/resourcesynccontroller/resourcesync_controller_test.go

```diff
@@ -111,3 +113,30 @@ func TestSyncConfigMap(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestServeHTTP(t *testing.T) {
+	c := &ResourceSyncController{
+		secretSyncRules: map[ResourceLocation]ResourceLocation{
+			{Namespace: "foo", Name: "cat"}: {Namespace: "bar", Name: "dog"},
```

Comment:
looks like your test needs multiple rules so that you would catch the unstable serialization.